pkg/controllers/management/rkeworkerupgrader/upgrade.go

```diff
@@ -374,6 +393,79 @@ func (uh *upgradeHandler) toUpgradeCluster(cluster *v3.Cluster) (bool, error) {
 	return false, nil
 }
 
+func (uh *upgradeHandler) restore(cluster *v3.Cluster) error {
+	nodes, err := uh.nodeLister.List(cluster.Name, labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	// allowed to use goroutine?
+	var errgrp errgroup.Group
+	nodesQueue := util.GetObjectQueue(nodes)
+	for w := 0; w < 5; w++ {
+		errgrp.Go(func() error {
+			var errList []error
+			for node := range nodesQueue {
+				node := node.(*v3.Node)
+				if node.Status.NodeConfig != nil && workerOnly(node.Status.NodeConfig.Role) {
+					if node.Status.NodePlan.Version != cluster.Status.NodeVersion {
+						if err := uh.processNode(node, cluster, false); err != nil {
```

Comment:
I think a comment should be added to `processNode` or it should be renamed to better clarify what it does. 
My understanding of what it does is:
* deep copies the node
* Sets the status nodePlan.Plan field
* Sets the status nodePlan.Version field 

It is hard to understand what / why those need to be set

Reply:
I had been using processNode during the upgrade logic consistently, as two stages of upgrade (preparing, then processing node) so missed that it has little context in this part of code 
- updated it to `setNodePlan`
- added a comment to restore function about why we need to update plan